src/tiled/objectselectiontool.cpp

```diff
@@ -825,6 +825,24 @@ void ObjectSelectionTool::updateHandleVisibility()
     for (int i = 0; i < AnchorCount; ++i)
         mResizeHandles[i]->setVisible(showHandles && mMode == Resize);
 
+    // Hide extra handles for the straight line polygons.
+    foreach (MapObject *mapObj, mapDocument()->selectedObjects()) {
+        if (mapObj->shape() == MapObject::Polyline &&
+            mapObj->polygon().size() == 2) {
+            mResizeHandles[TopLeftAnchor]->setVisible(false);
+            mResizeHandles[BottomLeftAnchor]->setVisible(false);
+            mResizeHandles[TopRightAnchor]->setVisible(false);
+            mResizeHandles[BottomRightAnchor]->setVisible(false);
```

Comment:
I'll take some tries again, so you can leave it to me and pay an attention to more important issues/features. Thanks for words, it seems I hurried with the patch.